client/core/trade.go

```diff
@@ -746,6 +759,12 @@ func (t *trackedTrade) swapMatches(matches []*matchTracker) error {
 		}
 	}
 
+	// Check if we need to return the change coin because the order has been
+	// canceled.
+	if t.metaData.Status == order.OrderStatusCanceled && !t.hasUnswappedMatches() {
+		t.wallets.fromWallet.ReturnCoins(asset.Coins{t.change})
+	}
```

Comment:
I think `(*trackedTrade).swapMatches` or `(*ExchangeWallet).Swap` also needs to at least try to unlock the funding coins it used as inputs, be they regular order coins or previous swap change.  Basically the `swaps.Inputs` slice.  They are now spent, so it doesn't matter much, but they seem to stay locked as far as listlockunspent is concerned.